src/infrastructure/gateway/payments/PaymentsInMemoryGateway.ts

```diff
@@ -0,0 +1,64 @@
+import IPaymentsGateway from "../../../domain/IPaymentsGateway";
+import { Payment, CreatePaymentRequest } from "@companieshouse/api-sdk-node/dist/services/payment";
+import { StatusCodes } from "http-status-codes";
+import { ApiResult, ApiResponse, ApiErrorResponse } from "@companieshouse/api-sdk-node/dist/services/resource";
+import { Failure, Success } from "@companieshouse/api-sdk-node/dist/services/result";
+
+class PaymentsInMemoryGateway implements IPaymentsGateway {
+
+  async createPayment(
+    opt: { access_token: string; refresh_token: string },
+    createPaymentRequest: CreatePaymentRequest) {
+    return await this.generatePaymentResult(opt, createPaymentRequest);
```

Comment:
don't think await is needed as there are no calls to async functions inside generatePaymentResult

Reply:
actually, i think i'm wrong as the interface says we have to return a promise so the await is probably needed